File: atm.c
#include <stdio.h>

#define C_RESET "\x1b[0m"
#define C_TEAL "\x1b[36m"
#define C_BLUE "\x1b[34m"
#define C_GREEN "\x1b[32m"
#define C_YELLOW "\x1b[33m"
#define C_RED "\x1b[31m"

typedef struct
{
    char card_code;
    const char *name;
    const char *title;
    int pin;
    int balance;
} Account;

static const Account accounts[] = {
    {'k', "Ken Kaneki", "Midnight Account", 1234, 50000},
    {'s', "Sasuke Uchiha", "Storm Account", 5678, 100000},
    {'i', "Itachi Uchiha", "Shadow Account", 9123, 60000},
    {'l', "Light Yagami", "Nova Account", 8123, 40000}};

static const int account_count = (int)(sizeof(accounts) / sizeof(accounts[0]));

static void print_bar(void)
{
    printf("+----------------------------------------------------------+\n");
}

static void print_header(void)
{
    print_bar();
    printf(C_TEAL "|                     ANTEIKU ATM v2                       |\n" C_RESET);
    printf(C_BLUE "|                Cool Neon Terminal Edition                |\n" C_RESET);
    print_bar();
    printf(" Card codes available: k / s / i / l\n\n");
}

static const Account *find_account_by_code(char code)
{
    int i;

    for (i = 0; i < account_count; i++)
    {
        if (accounts[i].card_code == code)
        {
            return &accounts[i];
        }
    }

    return NULL;
}

static int validate_pin(int expected_pin)
{
    int pin;
    int attempt;

    for (attempt = 1; attempt <= 3; attempt++)
    {
        printf(" Enter 4-digit PIN (attempt %d/3): ", attempt);
        if (scanf("%d", &pin) != 1)
        {
            return 0;
        }

        if (pin == expected_pin)
        {
            return 1;
        }

        printf(C_RED " PIN mismatch.\n" C_RESET);
    }

    return 0;
}

static void print_receipt(const Account *account, int withdrawn, int remaining)
{
    printf("\n");
    print_bar();
    printf(C_GREEN "|                    TRANSACTION RECEIPT                   |\n" C_RESET);
    print_bar();
    printf(" User        : %s\n", account->name);
    printf(" Profile     : %s\n", account->title);
    printf(" Withdrawn   : %d\n", withdrawn);
    printf(" Remaining   : %d\n", remaining);
    print_bar();
}

static void run_session(const Account *account)
{
    int choice;
    int amount;

    printf(C_GREEN "\n Welcome, %s\n" C_RESET, account->name);
    printf(" Profile: %s\n", account->title);
    print_bar();
    printf(" 1) Withdraw Cash\n");
    printf(" 2) Check Balance\n");
    print_bar();
    printf(" Choose action: ");

    if (scanf("%d", &choice) != 1)
    {
        printf(C_RED " Invalid action input.\n" C_RESET);
        return;
    }

    if (!validate_pin(account->pin))
    {
        printf(C_RED " Access blocked for this session.\n" C_RESET);
        return;
    }

    if (choice == 1)
    {
        printf(" Enter amount to withdraw: ");
        if (scanf("%d", &amount) != 1)
        {
            printf(C_RED " Invalid amount input.\n" C_RESET);
            return;
        }

        if (amount <= 0)
        {
            printf(C_RED " Amount must be greater than zero.\n" C_RESET);
            return;
        }

        if (amount > account->balance)
        {
            printf(C_RED " Insufficient funds.\n" C_RESET);
            return;
        }

        printf(C_GREEN "\n Cash ready. Please collect: %d\n" C_RESET, amount);
        print_receipt(account, amount, account->balance - amount);
    }
    else if (choice == 2)
    {
        printf(C_YELLOW "\n Current Balance: %d\n" C_RESET, account->balance);
    }
    else
    {
        printf(C_RED " Unknown menu action.\n" C_RESET);
    }
}

int main(void)
{
    char card_code;
    const Account *account;

    print_header();
    printf(" Insert card code: ");
    if (scanf(" %c", &card_code) != 1)
    {
        printf(C_RED " Failed to read card.\n" C_RESET);
        return 0;
    }

    account = find_account_by_code(card_code);
    if (account == NULL)
    {
        printf(C_RED " Unknown card code. Access denied.\n" C_RESET);
        print_bar();
        return 0;
    }

    run_session(account);

    printf("\n");
    print_bar();
    printf(C_TEAL " Session complete. Thanks for using ANTEIKU ATM.\n" C_RESET);
    print_bar();
    return 0;
}

File: atm.py
C_RESET = "\x1b[0m"
C_TEAL = "\x1b[36m"
C_BLUE = "\x1b[34m"
C_GREEN = "\x1b[32m"
C_YELLOW = "\x1b[33m"
C_RED = "\x1b[31m"

ACCOUNTS = [
    {"card_code": "k", "name": "Ken Kaneki",
        "title": "Midnight Account", "pin": 1234, "balance": 50000},
    {"card_code": "s", "name": "Sasuke Uchiha",
        "title": "Storm Account", "pin": 5678, "balance": 100000},
    {"card_code": "i", "name": "Itachi Uchiha",
        "title": "Shadow Account", "pin": 9123, "balance": 60000},
    {"card_code": "l", "name": "Light Yagami",
        "title": "Nova Account", "pin": 8123, "balance": 40000},
]


def print_bar():
    print("+----------------------------------------------------------+")


def print_header():
    print_bar()
    print(f"{C_TEAL}|                     ANTEIKU ATM v2                       |{C_RESET}")
    print(f"{C_BLUE}|                Cool Neon Terminal Edition                |{C_RESET}")
    print_bar()
    print(" Card codes available: k / s / i / l\n")


def find_account_by_code(code):
    for account in ACCOUNTS:
        if account["card_code"] == code:
            return account
    return None


def validate_pin(expected_pin):
    for attempt in range(1, 4):
        try:
            pin = int(input(f" Enter 4-digit PIN (attempt {attempt}/3): "))
        except ValueError:
            return False

        if pin == expected_pin:
            return True

        print(f"{C_RED} PIN mismatch.{C_RESET}")

    return False


def print_receipt(account, withdrawn, remaining):
    print()
    print_bar()
    print(f"{C_GREEN}|                    TRANSACTION RECEIPT                   |{C_RESET}")
    print_bar()
    print(f" User        : {account['name']}")
    print(f" Profile     : {account['title']}")
    print(f" Withdrawn   : {withdrawn}")
    print(f" Remaining   : {remaining}")
    print_bar()


def run_session(account):
    print(f"{C_GREEN}\n Welcome, {account['name']}{C_RESET}")
    print(f" Profile: {account['title']}")
    print_bar()
    print(" 1) Withdraw Cash")
    print(" 2) Check Balance")
    print_bar()

    try:
        choice = int(input(" Choose action: "))
    except ValueError:
        print(f"{C_RED} Invalid action input.{C_RESET}")
        return

    if not validate_pin(account["pin"]):
        print(f"{C_RED} Access blocked for this session.{C_RESET}")
        return

    if choice == 1:
        try:
            amount = int(input(" Enter amount to withdraw: "))
        except ValueError:
            print(f"{C_RED} Invalid amount input.{C_RESET}")
            return

        if amount <= 0:
            print(f"{C_RED} Amount must be greater than zero.{C_RESET}")
            return

        if amount > account["balance"]:
            print(f"{C_RED} Insufficient funds.{C_RESET}")
            return

        print(f"{C_GREEN}\n Cash ready. Please collect: {amount}{C_RESET}")
        print_receipt(account, amount, account["balance"] - amount)
    elif choice == 2:
        print(f"{C_YELLOW}\n Current Balance: {account['balance']}{C_RESET}")
    else:
        print(f"{C_RED} Unknown menu action.{C_RESET}")


def main():
    print_header()
    card_code = input(" Insert card code: ").strip().lower()

    if not card_code:
        print(f"{C_RED} Failed to read card.{C_RESET}")
        return

    account = find_account_by_code(card_code[0])
    if account is None:
        print(f"{C_RED} Unknown card code. Access denied.{C_RESET}")
        print_bar()
        return

    run_session(account)

    print()
    print_bar()
    print(f"{C_TEAL} Session complete. Thanks for using ANTEIKU ATM.{C_RESET}")
    print_bar()


if __name__ == "__main__":
    main()
